libipl/p10/ipl0.C

```diff
@@ -374,28 +387,124 @@ static int ipl_alignment_check(void)
 	return -1;
 }
 
+static int initialize_and_check_clock_chip(uint8_t& clock_count)
+{
+	struct pdbg_target *clock_target;
+	enum pdbg_target_status status;
+	uint8_t data;
+	int rc = 0;
+
+	// initialize output variable.
+	clock_count = 0;
+	
+	ipl_log(IPL_DEBUG, "Istep: soft reset clock, and verify clock status register\n");
+	pdbg_for_each_class_target("oscrefclk", clock_target) {
+		status = pdbg_target_probe(clock_target);
+		if(status != PDBG_TARGET_ENABLED){
```

Comment:
we are expecting all available clock always functional.
If i use, ipl_is_functional and it return flase, i may have to report it as software error and exit IPL.
Whether it will look nice, if i write code like this?

Reply:
> we are expecting all available clock always functional.

So, If any one of the clocks is failed then, we don't need to attempt to init other available clocks and just exit the IPL?

Reply:
If any clock showing not functional, that indicates there is a problem in software.

So, i think we should exit immediately without doing anything else.

It might be possible that there is a bug in devtree, and when we trying to soft reset to the second clock it may go to another device.

Reply:
i didnt get this.

Reply:
As per our design, clock is not a gardable target